sdk/search/Azure.Search.Documents/src/Utilities/AggregateDictionary.cs

```diff
@@ -0,0 +1,311 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Azure.Search.Documents
+{
+    /// <summary>
+    /// Aggregates multiple dictionaries as a single dictionary.
+    /// </summary>
+    /// <typeparam name="TKey">The type of key.</typeparam>
+    /// <typeparam name="TValue">The type of value.</typeparam>
+    internal abstract class AggregateDictionary<TKey, TValue> : IDictionary<TKey, TValue>
```

Comment:
I suspect implementation might be simpler if we make it less generic and more specific to the IndexingParametersConfiguration case. It would also be more efficient because you won't have to allocate 2 extra dictionaries.

Reply:
I'm not allocating 2 extra dictionaries. I'm allocating an array to reference to existing dictionaries. I'll also be redoing this slightly to avoid `IEnumerator<T>` allocations, reducing GC pressure and keeping with stack memory.

I originally started this making it specific to the case at hand, but if we're going to keep doing JSON objects with well-known properties and additional properties this way, I figured having an abstraction that handles these cases ready should be easier to consume. It also helped separate the logic and testability a bit, but could be convinced to just collapse them into a single definition.

Reply:
This is gone now.